src/actions/WrapWithSnippet.ts

```diff
@@ -0,0 +1,95 @@
+import { commands, workspace } from "vscode";
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  PartialTarget,
+  ScopeType,
+  SnippetName,
+  TypedSelection,
+} from "../typings/Types";
+import displayPendingEditDecorations from "../util/editDisplayUtils";
+import { ensureSingleEditor } from "../util/targetUtils";
+import { callFunctionAndUpdateSelections } from "../util/updateSelections";
+
+interface SnippetPreferences {
+  scopeType?: ScopeType;
+}
+
+type UserLanguageSnippetMap = Record<string, SnippetPreferences>;
+
+export default class WrapWithSnippet implements Action {
+  getTargetPreferences(
+    partialTargets: PartialTarget[],
```

Comment:
We have this arg here so that below FIXME can be implemented in the future.  Mixed feelings about letting action see this thing but not sure how else to let it do something like figure out which editor a target is referring to

Reply:
- [ ] pass in context object with callbacks instead